test-e2e.sh

```diff
@@ -81,6 +81,22 @@ kubectl version
 # Don't build/test the prow image as it requires Docker
 EXCLUDED_TARGETS="-//images/gcloud-bazel:gcloud_install -//images/gcloud-bazel:gcloud_push"
 
+# Create a unique namespace for this job using the repo name and the build id
+export E2E_NAMESPACE="build-${BUILD_ID:-0}"
+
+(cat <<EOF
+{
+  "kind": "Namespace",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "${E2E_NAMESPACE}",
+    "labels": {
+    }
+  }
+}
+EOF
+) |  kubectl create -f - || true
+
```

Comment:
Please try to minimize the chance of leaking namespaces:
```sh
cleanup() {
  # Delete everything in the namespace
  kubectl delete --all --namespace=$E2E_NAMESPACE
  # Delete the namespace
  kubectl delete namespaces/$E2E_NAMESPACE
}

# Clean up the namespace before leaving this script
trap "echo Cleaning up... && cleanup" EXIT
```

Will probably still want to have a janitor periodic prowjob that finds old namespaces and deletes them

Reply:
Added this. Will look into creating a janitor job as well